proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+
+**Acceptance Criteria:**
+
+- KFP artifacts use filesystem storage via `kfp-artifacts://` URIs
+- Components using boto3/gsutil for S3 continue to work
+- Can mix storage backends (system artifacts on PVC, custom data on S3)
+- Clear documentation on what uses filesystem vs object storage
+
+#### Story 5: Operator Deploying Multi-Tenant KFP with Namespace Isolation
+
+As an operator, I want to deploy KFP in namespace-local mode where each namespace annotated with `pipelines.kubeflow.org/enabled=true` gets its own artifact server pod and dedicated PVC, so that Team A's artifacts in namespace `team-a` are physically isolated from Team B's artifacts in namespace `team-b`.
+
+**Acceptance Criteria:**
+
+- Each namespace with `pipelines.kubeflow.org/enabled=true` annotation gets its own artifact server deployment
+- Each namespace gets its own dedicated PVC (no shared storage)
+- Artifact server in `team-a` namespace cannot access PVC in `team-b` namespace
+- Users can only access artifacts in namespaces they have RBAC permissions for (via `SubjectAccessReview`)
+- Physical isolation verified: deleting `team-a` namespace doesn't affect `team-b`'s artifacts
+
+#### Story 6: Operator Preferring KFP-Native Storage
+
+As an operator in a regulated environment (e.g., healthcare, finance), I want to deploy KFP with filesystem storage using an encrypted `StorageClass` (e.g., `encrypted-gp3`), so that artifact handling stays within the KFP codebase and I don't need to include a separate object storage system in my security audits.
+
+**Acceptance Criteria:**
+
+- All artifacts stored on PVCs within the cluster
+- KFP configuration uses `Filesystem.Type: "pvc"` with encrypted `StorageClass`
+- `SubjectAccessReview` validates all artifact access requests
+- Encryption at rest provided by the configured `StorageClass` (e.g., `encrypted-gp3`)
+- No separate object storage component to audit
+
+#### Story 7: Operator Running KFP on Storage-Constrained Infrastructure
+
+As an operator with limited storage budget (only 1TB total available), I want to deploy KFP in central mode with a single 500GB PVC shared across 10 team namespaces, so that all teams can run pipelines without each needing their own 100GB PVC (which would require 1TB total).
+
+**Acceptance Criteria:**
+
+- Central mode configured with: `DeploymentMode: "central"`, `Size: "500Gi"`
+- Single PVC created in `kubeflow` namespace mounted by one artifact server
+- All 10 teams' artifacts stored in `/artifacts/<namespace>/` directories on same PVC
+- Teams can run pipelines concurrently without storage allocation failures
+- No per-namespace storage limits (trade-off of central mode - shared PVC means no per-team quotas)
```

Comment:
```suggestion
```

That is also just fully covered by the default multi-tenant seaweedfs. Just one single pvc in the kubeflow namespace.